dashboard/app/views/home/index.html.haml

```diff
@@ -11,6 +11,8 @@
 -# Teacher modals
 - if current_user && current_user.teacher? && !current_user.accepted_latest_terms?
   = render partial: 'layouts/terms_interstitial'
+- elsif (current_user && current_user.show_school_info_confirmation_dialog?) || @force_school_info_confirmation_dialog
```

Comment:
It looks like `@force_school_info_confirmation_dialog` isn't set anywhere - you probably need to do this in the `HomeController`.